handler/src/main/java/io/netty/handler/ssl/TlsMasterKeyBiConsumer.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import javax.crypto.SecretKey;
+
+/**
+ * An interface to implement that will be called via {@link TlsMasterKeyHandler} after
+ * {@link SslHandshakeCompletionEvent} with the contents of the master key for the session.
+ * This can be extremely useful in <b>debug-only</b> scenarios if you ever want to decrypt the TLS traffic.
+ */
+public interface TlsMasterKeyBiConsumer {
```

Comment:
imho we could just keep things simple and not adding this interface and just provide protected methods in `TlsMasterKeyHandler` that people override and make the class abstract.